### DISCUSSION TITLE
python-socketio doesn't work with FastAPI 0.109

### DISCUSSION THREAD

**Discussion by karel-majer** [fastapi/fastapi#10970] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware
import socketio

app = FastAPI()
sio = socketio.AsyncServer(async_mode="asgi", cors_allowed_origins=[])
sio_app = socketio.ASGIApp(sio)
app.mount("/ws", sio_app)

app.add_middleware(
    CORSMiddleware, allow_origins=["*"], allow_methods=["*"], allow_headers=["*"]
)
```

### Description

The app above is run by uvicorn. With fastapi<=0.108 the app works and GET to http://localhost:8000/ws/socket.io/?EIO=4&transport=polling returns something like:
0{"sid":"8hEAy539LcpMSgfMAAAA","upgrades":["websocket"],"pingTimeout":20000,"pingInterval":25000}
With fastapi==0.109 I get 404 Not Found instead.

Is there any change in how socketio application should be mounted?

### Operating System

Linux, Windows

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.11.3

**Comment by Kludex** [comment, marked as answer]
Discussion: https://github.com/encode/starlette/discussions/2413

**Reply by karel-majer** [reply]
Thank you, that was it. Solved by adding socketio_path parameter to ASGIApp like this:
sio_app = socketio.ASGIApp(sio, socketio_path="/ws/socket.io")
